Wba single wire uart

lkoudelka opened the pull request:
This pull request introduces support for the Nucleo WBA55CG board in the single-wire UART example.

erwango commented on samples/boards/st/uart/single_wire/README.rst:
```diff
@@ -12,6 +12,8 @@ functionality of STM32. Without adaptions this example runs on STM32F3 discovery
 board. You need to establish a physical connection between pins PA2 (USART2_TX) and
 PC10 (UART4_TX).
 
+To support the `nucleo_wba55cg` board, establish a physical connection between pins PA12 (USART2_TX) and PA2 (LPUART1_TX).
+
```
I would club this into a single description pushing readers to go to each board overlay to see which connection should be made, in order to avoid the need to modify this each time a new board is added

erwango commented on samples/boards/st/uart/single_wire/boards/nucleo_wba55cg.overlay:
```diff
@@ -0,0 +1,31 @@
+// nucleo_wba55cg_single_wire_uart.overlay
```
Please use usual file header with copyright

erwango commented on samples/boards/st/uart/single_wire/boards/nucleo_wba55cg.overlay:
```diff
@@ -0,0 +1,31 @@
+	status = "okay";
+	pinctrl-0 = <&usart2_tx_pa12>;
+	current-speed = <115200>;
+	single-wire;  // Enable single-wire mode
```
Avoid C++ style comments (`//`)

erwango commented on samples/boards/st/uart/single_wire/boards/nucleo_wba55cg.overlay:
```diff
@@ -0,0 +1,31 @@
+
+/ {
+	aliases {
+	singlelineuart1 = &usart2;
+	singlelineuart2 = &lpuart1;
```
Please fix indentation
```suggestion
		singlelineuart1 = &usart2;
		singlelineuart2 = &lpuart1;
```

rustyh1 replied:
rename 
singlelineuart1 --> single-line-uart1
singlelineuart2 --> single-line-uart2
to match the main.c file:
...
#define UART_NODE1 DT_ALIAS(single_line_uart1)
#define UART_NODE2 DT_ALIAS(single_line_uart2)
...

erwango commented on samples/boards/st/uart/single_wire/boards/stm32f3_disco.overlay:
```diff
@@ -6,8 +6,8 @@
 
 / {
 	 aliases {
-		single-line-uart1 = &usart2;
-		single-line-uart2 = &uart4;
```
I don't think this was the actual error. `-` in aliases are used widely.

lkoudelka committed:
boards: st: nucleo_wba55cg: Add USART2 definition

Added the USART2 definition to the nucleo_wba55cg device tree source file.
This enables the use of USART2 on the nucleo_wba55cg board.

Signed-off-by: Lubos Koudelka <lubos.koudelka@st.com>

lkoudelka committed:
samples: uart: single_wire: Add support for nucleo_wba55cg

Added support for the nucleo_wba55cg board to the uart/single_wire example.
- Updated the overlay file to configure USART2 and LPUART1 for single-wire
  mode.
- Modified sample.yaml to include nucleo_wba55cg in the list of supported
  boards.
- Updated README.rst to provide generic, board-agnostic instructions for
  building and running the application.

Signed-off-by: Lubos Koudelka <lubos.koudelka@st.com>

erwango commented on samples/boards/st/uart/single_wire/README.rst:
```diff
@@ -4,13 +4,16 @@
 
-Overview
-********
+# Overview
```
Please keep rst formatting.
See https://builds.zephyrproject.io/zephyr/pr/82946/docs/samples/boards/st/uart/single_wire/README.html#uart-stm32-single-wire for the rendering of this change.

erwango reviewed (changes requested):
Otherwise LGTM

rustyh1 reviewed (commented):
Sounds good for me

erwango commented on samples/boards/st/uart/single_wire/README.rst:
```diff
@@ -7,14 +7,18 @@
 Add a ``single_wire_uart_loopback`` fixture to your board in the hardware map to allow
 twister to verify this sample's output automatically.
 
+Hardware Setup
+********
```
```suggestion
Hardware Setup
**************
```

Required to solve https://github.com/zephyrproject-rtos/zephyr/actions/runs/13160412192/job/36727517154?pr=82946#step:8:66

djiatsaf-st reviewed (commented):
Otherwise LGTM
